qt/applications/workbench/workbench/plotting/figuremanager.py

```diff
@@ -349,47 +347,52 @@ def _set_fit_enabled(self, on):
         action.setEnabled(on)
         action.setVisible(on)
 
+    def generate_plot_script_clipboard(self):
+        script = generate_script(self.canvas.figure, exclude_headers=True)
+        QApplication.clipboard().setText(script)
+        logger.notice("Plotting script copied to clipboard.")
+
+    def generate_plot_script_file(self):
+        script = generate_script(self.canvas.figure)
+        filepath = open_a_file_dialog(
+            parent=self.canvas,
+            default_suffix=".py",
+            file_filter="Python Files (*.py)",
+            accept_mode=QFileDialog.AcceptSave,
+            file_mode=QFileDialog.AnyFile
+        )
+        if filepath:
+            try:
+                with open(filepath, 'w') as f:
+                    f.write(script)
+            except IOError as io_error:
+                logger.error("Could not write file: {}\n{}"
+                             "".format(filepath, io_error))
+
+    def _set_generate_plot_script_enabled(self, enabled):
+        action = self.toolbar._actions['generate_plot_script']
+        action.setEnabled(enabled)
+        action.setVisible(enabled)
+        # Also hide the separator between this and "Fit" button
+        for i, toolbar_action in enumerate(self.toolbar.actions()):
+            if toolbar_action == action:
+                self.toolbar.actions()[i+1].setVisible(enabled)
```

Comment:
```suggestion
                self.toolbar.actions()[i+1].setVisible(not enabled)
```
If this is trying to hide the separator should it do `setVisible` with the opposite of the visibility of the generate button? Maybe the comment above is misleading as it would make more sense to show the separator when the button is visible (which is the current behaviour)

Reply:
Yep the comment is bad!  

Is this better?
```python
# Also show/hide the separator ...
```

Reply:
Yeah O think that's better